syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "custom_types.proto";

service AccountService {
  rpc GetAll (GetAllRequest) returns (GetAllAccountsResponse);
  rpc GetById (GetAccountByIdRequest) returns (AccountModel);
  rpc GetByCustomerId (GetAccountByCustomerIdRequest) returns (GetAllAccountsResponse);
  rpc GetByDateRange (GetByDateRangeRequest) returns (GetAllAccountsResponse);
}

message AccountModel {
  int32 account_id = 1;
  optional int32 customer_id = 2;
  optional int32 type_id = 3;
  string iban = 4;
  optional string balance = 5;
  optional google.protobuf.Timestamp created_at = 6;
  optional bool is_active = 7;
}

message GetAccountByCustomerIdRequest {
  int32 customer_id = 1;
}

message GetAllAccountsResponse {
  repeated AccountModel accounts = 1;
}

message GetAccountByIdRequest {
  int32 account_id = 1;
}
